### ISSUE TITLE
When populating a Tree or a TreeGrid, I want to be able to construct the data without accessing TreeData for simplicity

### ISSUE THREAD

**Issue by hesara** [vaadin/framework#9374]
Provide API on Tree/TreeGrid level for easily constructing in-memory content.

Based on feedback from @mstahv .

**Comment by Legioth** [comment]
Constructing data is already doable through `setItems(Collection<T> rootItems, ValueProvider<T, Collection<T>> childItemProvider) `.

Was this ticket based on the discussion about being able to also modify the data directly through e.g. `TreeGrid` instead of doing `getTreeData()`?

**Comment by hesara** [comment]
Yes, I believe the idea was to be able to construct or manipulate data through an API on the component itself. This approach has some issues as the API cannot apply to all instances with an arbitrary DataProvider, so I guess the setItems() mentioned by Leif (already implemented) is the best we can do right now.
